xray/test/test_backends.py

```diff
@@ -651,6 +658,24 @@ def roundtrip(self, data, **kwargs):
                 yield ds
 
 
+@requires_netCDF4
+class NetCDF4ClassicViaNetCDF4DataTest(CFEncodedDataTest, TestCase):
```

Comment:
Instead of making a new test case for netCDF4 classic, what about writing an explicit test that dtypes are preserved when writing a dataset using this mode? e.g., something like the following:
```
# can add this on a new TestCase class, with @requires_netCDF4
def test_netcdf_dtypes(self):
    expected = Dataset(...)  # something with float64 data
    for format in ['netcdf4_classic', 'netcdf4', 'netcdf3_classic', 'netcdf3_64bit']:
        with create_tmp_file() as tmp_file:
            expected.to_netcdf(tmp_file, format=format)
            with open_dataset(tmp_file) as actual:
                self.assertDatasetIdentical(expected, actual)
```